fitwsarah_frontend/src/views/AdminPanelPage/AdminAppointments.js

```diff
@@ -122,43 +221,117 @@ function AdminAccounts() {
+                                            <input
+                                                type="text"
+                                                name="phoneNum"
+                                                value={editFormData.phoneNum}
+                                                onChange={handleChange}
+                                            />
+                                        </td>
+                                        <td className="date-time-picker-column">
+                                            <LocalizationProvider dateAdapter={AdapterDateFns}>
+                                                <DateTimePicker
+                                                    label="Date & Time picker"
+                                                    value={editFormData.dateTime}
+                                                    onChange={handleDateTimeChange}
+                                                    renderInput={(params) => <TextField {...params} />}
+                                                />
+                                            </LocalizationProvider>
+                                        </td>
+                                        <td className="button-container">
+                                            <button className="saveButton" onClick={() => handleSaveClick(appointment.appointmentId)}>
+                                                Save
+                                            </button>
+                                            <button className="cancelButton" onClick={() => setEditAppointmentId(null)}>
+                                                Cancel
+                                            </button>
+                                        </td>
+                                    </tr>
+                                ) : (
+                                    <tr key={appointment.appointmentId}>
+                                        <td>{appointment.appointmentId}</td>
+                                        <td>{appointment.status}</td>
+                                        <td>{appointment.location}</td>
+                                        <td>{appointment.firstName}</td>
+                                        <td>{appointment.lastName}</td>
+                                        <td>{appointment.phoneNum}</td>
+                                        <td>{appointment.date}</td>
+                                        <td>{appointment.time}</td>
+                                        <td className="edit-button-container">
+                                            <button className="saveButton" onClick={() => handleEditClick(appointment)}>
+                                                Edit
+                                            </button>
+                                            <button className="cancelButton" onClick={() => handleCancelAppointment(appointment.appointmentId)}>
```

Comment:
why does this need the appointment Id but the handleEditClick function does not?

Reply:
That part needs the id because it's getting the actual Id of every appointment to show them on the screen. As for the edit form the reason why it doesn't use the id is because for each appointment there's an edit button so when you click an edit button and it brings up the form for it the system knows it's for that specific appointment.